app/cards/renaissance/border_guard.js

```diff
@@ -0,0 +1,160 @@
+            return card.name === selected_card.name
+        })
+
+        player_cards.revealed.splice(selected_card_index, 1)
+
+        player_cards.hand.push(selected_card)
+        game.log.push(`&nbsp;&nbsp;<strong>${player_cards.username}</strong> puts ${CardView.render(selected_card)} in their hand`)
+
+        let card_discarder = new CardDiscarder(game, player_cards, 'revealed', _.map(player_cards.revealed, 'name'))
+        card_discarder.discard()
+
+        if (are_all_actions) {
+            let turn_event_id = TurnEventModel.insert({
+                game_id: game._id,
+                player_id: player_cards.player_id,
+                username: player_cards.username,
+                type: 'choose_options',
+                instructions: `Choose One:`,
+                minimum: 1,
+                maximum: 1,
+                options: [
+                    { text: 'Take the Horn', value: 'horn' },
+                    { text: 'Take the Lantern', value: 'lantern' }
+                ]
+            })
+
+            let turn_event_processor = new TurnEventProcessor(game, player_cards, turn_event_id)
+            turn_event_processor.process(BorderGuard.process_response)
+        }
+    }
+
+    static process_response(game, player_cards, response) {
+        response = response[0]
+        if (response === 'horn') {
+            BorderGuard.get_artifact(game, player_cards, new Horn(), 'Horn')
+        } else if (response === 'lantern') {
+            BorderGuard.get_artifact(game, player_cards, new Lantern(), 'Lantern')
+        }
+    }
+
+    static get_artifact(game, player_cards, artifact_instance, artifact_name) {
+        let artifact_index = _.findIndex(player_cards.artifacts, function (artifact) {
+            return artifact.name === artifact_name
+        })
+
+        if (artifact_index === -1) {
+            let ordered_player_cards = TurnOrderedPlayerCardsQuery.turn_ordered_player_cards(game, player_cards)
+            ordered_player_cards.shift()
+
+            _.each(ordered_player_cards, function (other_player_cards) {
+                let other_player_artifact_index = _.findIndex(other_player_cards.artifacts, function (artifact) {
+                    return artifact.name === artifact_name
+                })
+                if (other_player_artifact_index !== -1) {
+                    other_player_cards.artifacts.splice(other_player_artifact_index, 1)
+                }
+                PlayerCardsModel.update(game._id, other_player_cards)
+            })
+
+            let artifact = artifact_instance.to_h()
+            player_cards.artifacts.push(artifact)
+            game.log.push(`&nbsp;&nbsp;<strong>${player_cards.username}</strong> takes ${CardView.render(artifact, true)}`)
+        }
+    }
+
+    discard_event(discarder) {
+        let game = discarder.game
+        let player_cards = discarder.player_cards
+
+        if (game.turn.horns > 0) {
```

Comment:
Given that there are several conditions that need to be met before this event fires off, I'd prefer to see that conditional logic implemented in `discard_event_processor.js`. With the way things are structured right now, I believe that even if the discard event will not have an effect, it'll still potentially prompt the user to resolve the event.